src/models.js

```diff
@@ -0,0 +1,76 @@
+'use strict'
+
+//------------------------------------------------------------------------------
+// Models for our locally stored data
+//------------------------------------------------------------------------------
+var Sequelize = require('sequelize');
+var Q = require('q');
+var AsyncLock = require('async-lock');
+
+
+var exports = (function() {
+    var _models = {};
+    var _lockQueue = [];
+    var _lock = new AsyncLock();
+
+    //--------------------------------------------------------------------------
+    // A lock that we can use to ensure that any writing to the database is 
+    // finished before another writer starts.
+    //
+    // A donePromise will be passed into the deferred, this donePromise should
+    // always be resolved when you are done with the database.
+    //--------------------------------------------------------------------------
+    function lock() {
+        var lockDeferred = Q.defer();
+        _lock.acquire("chesster", function() {
+            var unlock = Q.defer();
+            lockDeferred.resolve(unlock);
+            return unlock.promise;
+        });
+        return lockDeferred.promise;
+    }
+
+    //--------------------------------------------------------------------------
+    function defineModels(sequelize) {
+        module.exports.LichessRating = sequelize.define('LichessRating', {
```

Comment:
How/why does the definition of this model differ from the database migrations? It seems the model does not expose creation and update details. 

Will a separate model contain this information? Some audit model?